### ISSUE TITLE
other use case; generic approach

### ISSUE THREAD

**Issue by x0rzkov** [ddxgz/linkedinfo-ml#9]
Hi,

Hope you are all well !

I just wanted to ask you if your approach could apply to classify arxiv abstracts into specific categories/tasks ?

More precisely, I have a dataset of 90k abstracts classified in tasks.
https://paper2code.com/public/tag_predict.txt.tar.gz

How can I train a model based on this dataset ?

Cheers,
X

**Comment by ddxgz** [comment]
Hi,

What I performed is a typical multi-label classification. I had a quick look at the data you have, there seems no label to the abstracts. If you want to do a classification of the arxiv abstracts, you need to have a labeled dataset. If you don't have labeled dataset, you may have to do it in an unsupervised way. You may look into methods like topic modeling, using Transformer-based model to get representation of the abstracts to do a clustering, maybe combined with keywords extraction or so. 

The topics of the papers in your dataset seem all about machine learning, so I guess it might not be easy to get a satisfied classification result with unlabeled data.

**Comment by ddxgz** [comment]
Sorry I just found that there's a `tasks` column in your dataset. Is it the categories you want to classify? If it is, then you can simply use `tasks` as labels to do a multi-label classification.

**Comment by x0rzkov** [comment]
yes, tasks are the categories I want to classify.

How do you train the model ? could not find any doc or explanation about it ?

**Comment by x0rzkov** [comment]
DO you have a twitter account ? I would like to ask you some questions in DM and not polluting this feed ^^.

My twitter is https://twitter.com/x0rzkov

**Comment by ddxgz** [comment]
The training process of this model is nothing special, it is documented in the 2 posts in README. I think you can find many other detailed posts on training a text classifier using either traditional like tf-idf, or Transformer-based pretrained models. 

I'll connect on twitter, if you have more questions.

**Comment by x0rzkov** [comment]
thanks !

**Comment by x0rzkov** [comment]
@ddxgz 
I wanted to run your notebook but I cannot find the "infos_0_3353_fulltext.json" file.

Sorry, could not find you on twitter. Sorry for polluting this closed issue ^^.

**Comment by ddxgz** [comment]
`infos_0_3353_fulltext.json` is my dataset, which is not included in this repo, but can be retrieved by the script in this repo.

I've followed you and sent you a DM on twitter.

**Comment by x0rzkov** [comment]
Replied to you, thanks :-)
